### PR TITLE
[ci] Add build docker test-parachain for zombienet

### PR DESCRIPTION
**Pull request by alvicsam**
PR adds building and publishing `test-parachain` docker image

Close https://github.com/paritytech/ci_cd/issues/436

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -171,19 +171,36 @@ build-linux-stable:
...
   image:                           quay.io/buildah/stable
   needs:
     - job:                         build-linux-stable
       artifacts:                   true
+  # Variables here are just examples, override them in a job
   variables:
-    DOCKERFILE:                    "docker/polkadot-parachain-debug_unsigned_injected.Dockerfile"
-    IMAGE_NAME:                    docker.io/paritypr/polkadot-parachain-debug
+    DOCKERFILE:                    "docker/path-to.Dockerfile"
+    IMAGE_NAME:                    docker.io/paritypr/image_name
```
[the remark this answers is not in the document]
Comment (sergejparity) [reply .gitlab-ci.yml:203]: Oh. I see `A3-ingrogress` :)

**Review by rcny — APPROVED** [review]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -171,19 +171,32 @@ build-linux-stable:
...
+.build-push-image:                  &build-push-image
   image:                           quay.io/buildah/stable
-  needs:
-    - job:                         build-linux-stable
-      artifacts:                   true
   variables:
-    DOCKERFILE:                    "docker/polkadot-parachain-debug_unsigned_injected.Dockerfile"
-    IMAGE_NAME:                    docker.io/paritypr/polkadot-parachain-debug
+    DOCKERFILE:                    "" # docker/path-to.Dockerfile
+    IMAGE_NAME:                    "" # docker.io/paritypr/image_name
```
Comment (TriplEight) [inline .gitlab-ci.yml:199]: why do you need these variables at all here if there are no default values? As a reminder to set them?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -202,6 +215,32 @@ build-push-image:
...
+  stage:                           publish
+  <<:                              *kubernetes-env
+  <<:                              *common-refs
+  <<:                              *build-push-image
+  needs:
+    - job:                         build-linux-stable
+      artifacts:                   true
+  variables:
+    DOCKERFILE:                    "docker/polkadot-parachain-debug_unsigned_injected.Dockerfile"
+    IMAGE_NAME:                    docker.io/paritypr/polkadot-parachain-debug
```
Comment (TriplEight) [inline .gitlab-ci.yml:228]: please decide, quoted or unqoted. For YAML it's whatever, just looks weird.

**Review by TriplEight — CHANGES_REQUESTED** [review]
small nits

File: docker/test-parachain_injected.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# metadata
+ARG VCS_REF
+ARG BUILD_DATE
+ARG IMAGE_NAME
+
+LABEL io.parity.image.authors="devops-team@parity.io" \
+	io.parity.image.vendor="Parity Technologies" \
+	io.parity.image.title="${IMAGE_NAME}" \
+	io.parity.image.description="Test parachain for Zombienet" \
+	io.parity.image.source="https://github.com/paritytech/cumulus/blob/${VCS_REF}/scripts/docker/test-parachain_injected.Dockerfile" \
```
Comment (TriplEight) [inline docker/test-parachain_injected.Dockerfile:12]: that's a wrong URL

File: .gitlab-ci.yml (hunk 2 above)
Comment (alvicsam) [reply .gitlab-ci.yml:199]: Exactly and examples for them

**Review by TriplEight — APPROVED** [review]

**Pull request merged**